src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -2173,6 +2173,22 @@ bool Compiler::optRedundantRelop(BasicBlock* const block)
     if (!usedCopy)
     {
         fgRemoveStmt(block, candidateStmt);
+
+        // Make sure the removed store node isn't referenced in any SSA definitions
+        assert(candidateStmt->GetRootNode()->OperIs(GT_STORE_LCL_VAR));
+        GenTreeLclVarCommon* const rootNode = candidateStmt->GetRootNode()->AsLclVarCommon();
+
+        LclVarDsc* const varDsc = lvaGetDesc(rootNode);
```

Comment:
Since the local is in SSA you should be able to locate the `LclSsaVarDsc` directly. Something like:
```C++
unsigned const ssaNum = rootNode->GetSsaNum();
LclSsaVarDsc* const defDsc = varDsc->GetPerSsaData(ssaNum);
assert(defDsc->GetDefNode() == rootNode);
defDsc->SetDefNode(nullptr);
```

Reply:
Thanks, fixed